### PR TITLE
Genesis Builder migration functionality

### PR DESCRIPTION
**Pull request by viquezclaudio**
Vesting and HTLC support

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: genesis-builder/src/config.rs [hunk 1]
```diff
@@ -65,6 +72,40 @@ pub struct GenesisAccount {
...
+pub struct GenesisHTLC {
+    #[serde(deserialize_with = "deserialize_nimiq_address")]
+    pub address: Address,
+    #[serde(deserialize_with = "deserialize_nimiq_address")]
+    pub sender: Address,
+    #[serde(deserialize_with = "deserialize_nimiq_address")]
+    pub recipient: Address,
+    #[serde(deserialize_with = "deserialize_coin")]
+    pub balance: Coin,
+
```
Comment (jsdanielh) [inline genesis-builder/src/config.rs]: Remove extra space

File: genesis-builder/src/lib.rs [hunk 2]
```diff
@@ -187,6 +196,46 @@ impl GenesisBuilder {
                 .expect("Failed to store account");
         }
 
+        debug!("Vesting contracts");
```
Comment (jsdanielh) [inline genesis-builder/src/lib.rs:199]: Is this needed?

File: genesis-builder/src/lib.rs [hunk 3]
```diff
@@ -187,6 +196,46 @@ impl GenesisBuilder {
...
+                total_amount: vesting_contract.total_amount,
+            });
+
+            accounts
+                .tree
+                .put(&mut txn, &key, account)
+                .expect("Failed to store account");
+        }
+
+        debug!("HTLC contracts");
```
Comment (jsdanielh) [inline genesis-builder/src/lib.rs:218]: Is this needed?

File: genesis-builder/src/config.rs [hunk 4]
```diff
@@ -65,6 +72,40 @@ pub struct GenesisAccount {
...
+    pub start_time: u64,
+    pub time_step: u64,
+    #[serde(deserialize_with = "deserialize_coin")]
+    pub step_amount: Coin,
+    #[serde(deserialize_with = "deserialize_coin")]
+    pub total_amount: Coin,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct GenesisHTLC {
```
Comment (jsdanielh) [inline genesis-builder/src/config.rs:101]: Could you add `rustdoc` to the `struct` and its members?

File: genesis-builder/src/config.rs [hunk 5]
```diff
@@ -65,6 +72,40 @@ pub struct GenesisAccount {
     pub balance: Coin,
 }
 
+#[derive(Clone, Debug, Deserialize)]
+pub struct GenesisVestingContract {
```
Comment (jsdanielh) [inline genesis-builder/src/config.rs:77]: Could you add `rustdoc` to the `struct` and its members?

**Review by jsdanielh — CHANGES_REQUESTED** [review]
The `toml` change needs to go to `scripts/devnet/templates/dev-albatross-genesis.toml.j2` as well

File: genesis-builder/src/lib.rs (hunk 2 above)
Comment (viquezclaudio) [reply genesis-builder/src/lib.rs:199]: No, but thats to be consistent with the other logs that were already in the file

File: genesis-builder/src/config.rs [hunk 6]
```diff
@@ -65,6 +72,57 @@ pub struct GenesisAccount {
...
+    pub step_amount: Coin,
+    /// Vesting contract total amount
+    #[serde(deserialize_with = "deserialize_coin")]
+    pub total_amount: Coin,
+}
+
+/// Struct that represents an HTLC in the toml file that is used to generate the genesis
+#[derive(Clone, Debug, Deserialize)]
+pub struct GenesisHTLC {
+    /// Vesting contract account address
```
Comment (jsdanielh) [inline genesis-builder/src/config.rs]:
```suggestion
    /// HTLC contract account address
```

**Review by jsdanielh — CHANGES_REQUESTED** [review]

**Review by jsdanielh — COMMENTED** [review]
Could you sign the commits?

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**